### ISSUE TITLE
New Strapi Deployment Crashing on Startup (ECONNREFUSED)

### ISSUE THREAD

**Issue by ronen25** [railwayapp-templates/strapi#6]
Hi,

Tried deploying Strapi using the built-in template. The repo is created correctly, and the service builds correctly in the Railway dashboard.

However, the deployment fails to start. In the deployment logs, I see the following message repeated a couple of times:
```
$ strapi start
[2022-10-13 18:42:09.105] debug: ⛔️ Server wasn't able to start properly.
[2022-10-13 18:42:09.107] error: connect ECONNREFUSED 127.0.0.1:5432
Error: connect ECONNREFUSED 127.0.0.1:5432
at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1247:16)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I have no other services/DBs other than a Next.js app.

Steps to Reproduce
-----
1. Select "New" from the dashboard and add the Strapi Template.
2. The repository is created correctly in my Github (https://github.com/ronen25/strapi)
3. Notice it builds correctly.
4. Deployment log begins to show the aforementioned message, and the deployment fails afterwards.

Thanks for the help and the great work! I'm really enjoying Railway!

**Comment by jsegal205** [comment]
@ronen25 I just went through this and if you are deploying a v4 version of Strapi, in the [config/database.js](https://github.com/ronen25/strapi/blob/main/config/database.js#L4-L11) sets up this following db connection info - 
```
connection: {
  host: env('DATABASE_HOST', '127.0.0.1'),
  port: env.int('DATABASE_PORT', 5432),
  database: env('DATABASE_NAME', 'strapi'),
  user: env('DATABASE_USERNAME', 'strapi'),
  password: env('DATABASE_PASSWORD', 'strapi'),
  schema: env('DATABASE_SCHEMA', 'public'), // Not required
  ssl: env.bool('DATABASE_SSL', false),
},
```

I resolved by adding the specific environment variables (`DATABASE_HOST`, `DATABASE_PORT`, `DATABASE_NAME`, `DATABASE_USERNAME`, `DATABASE_PASSWORD`) to the strapi service in Railway.

Or you could use [pg-connection-string](https://www.npmjs.com/package/pg-connection-string) to parse the provided postgres information that is created with the template but that won't work on a new template creation.

**Comment by ronen25** [comment]
@jsegal205 thanks for the comment!

I've recreated the Strapi template, and it looks like now, the Postgres database was created properly - I don't recall it being created before, meaning I only had the Strapi service - and everything works without needing to do additional configuration.

Not sure what was fixed (or if it was just a problem with my already-existing project, which already had a Postgres DB set up), but it looks OK now, so I'm closing the issue.
